### PR TITLE
add gh

### PR DESCRIPTION
**Pull request by rluria14**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rluria14** [comment]
@oblute @benhuff agree to be maintainers?

**Comment by scopatz** [comment]
@__mvn__ didn't you add this already?

**Comment by rluria14** [comment]
@scopatz it does appear that a PR was made about a month ago (slightly different recipe): https://github.com/conda-forge/staged-recipes/pull/10849. But, it looks like windows is not passing on that PR.

File: recipes/gh/bld.bat [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+:: Change to directory with main.go
+cd cmd\gh
+
+:: Build
+go build -v -o %PKG_NAME%.exe .
+
+:: Install Binary into %PREFIX%\bin
+mkdir -p %PREFIX%\bin
+mv %PKG_NAME% %PREFIX%\bin\%PKG_NAME%
```
Comment (scopatz) [inline recipes/gh/bld.bat]: This should have `if errorlevel 1 exit 1` after every line.  Example at https://github.com/conda-forge/make-feedstock/blob/master/recipe/bld.bat

File: recipes/gh/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('go') }}
+
+test:
+  requires:         # [win]
+    - m2-bash       # [win]
+    - m2-coreutils  # [win]
```
Comment (scopatz) [inline recipes/gh/meta.yaml]: I think you can get rid of these lines

File: recipes/gh/bld.bat (hunk 1 above)
Comment (rluria14) [reply recipes/gh/bld.bat]: changes made

File: recipes/gh/meta.yaml (hunk 2 above)
Comment (rluria14) [reply recipes/gh/meta.yaml]: reqs removed.

File: recipes/gh/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  license_file: LICENSE
+  summary: "Github CLI."
+  doc_url: https://cli.github.com/
+  dev_url: https://github.com/cli/cli
+
+extra:
+  recipe-maintainers:
+    - rluria14
+    - benhuff
+    - oblute
```
Comment (scopatz) [inline recipes/gh/meta.yaml]: @rluria14 @benhuff

Comment (rluria14) [reply recipes/gh/meta.yaml]: he's been removed as a maintainer

**Pull request merged**